File: 6_kyu/Casino_chips.go
package kata

import "sort"

func Solve(arr []int) int {
	days := 0
	sort.Ints(arr)
	for arr[1] > 0 {
		n := arr[1]
		if arr[0] > 0 {
			n -= arr[0] - 1
		}
		days += n
		arr[1] -= n
		arr[2] -= n
		sort.Ints(arr)
	}
	return days
}

File: 6_kyu/Casino_chips.rs
fn solve(arr: &[u32; 3]) -> u32 {
    let max_pile: u32 = *arr.iter().max().unwrap();
    let sum_piles: u32 = arr.iter().sum::<u32>();
    let sum_other_piles: u32 = sum_piles - max_pile;
    if sum_other_piles < max_pile {
        sum_other_piles
    } else {
        sum_piles / 2
    }
}
